middleware/kubernetes/handler.go

```diff
@@ -87,7 +87,7 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 	}
 
 	if len(records) == 0 {
-		return middleware.BackendError(&k, zone, dns.RcodeSuccess, state, nil /*debug*/, nil, middleware.Options{})
+		return middleware.BackendError(&k, zone, dns.RcodeNameError, state, nil /*debug*/, nil, middleware.Options{})
```

Comment:
NameErrorr should be handled by the k.IsNameError() above, but k8s already returns `false` for that. I.e. that would be the proper fix.